hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsencePeriodTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRAbsence_BAO_HRAbsencePeriod as AbsencePeriod;
+use CRM_HRAbsence_Test_Fabricator_HRAbsencePeriod as AbsencePeriodFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsencePeriodTest
+ *
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsencePeriodTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface, TransactionalInterface {
+
+  /**
+   * @var array
+   *  An array of AbsencePeriods created by the setUp() method
+   */
+  private $absencePeriods = [];
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()->installMe(__DIR__)->apply();
+  }
+
+  public function setUp() {
+    $absencePeriodTable = AbsencePeriod::getTableName();
+    // We need to disable fk checks in order to avoid constraint
+    // errors when truncating the absence period table
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 0;");
+    CRM_Core_DAO::executeQuery("TRUNCATE $absencePeriodTable");
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 1;");
+
+    $this->absencePeriods[] = AbsencePeriodFabricator::fabricate([
+      'name' => 'Period 1',
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
+    ]);
+
+    $this->absencePeriods[] = AbsencePeriodFabricator::fabricate([
+      'name' => 'Period 2',
+      'start_date' => CRM_Utils_Date::processDate('2017-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2017-12-31'),
+    ]);
+
+    $this->absencePeriods[] = AbsencePeriodFabricator::fabricate([
+      'name' => 'Period 3',
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+  }
+
+  public function testGetAbsencePeriodWithoutDatesReturnsAllTheAbsencePeriods() {
+    $periods = AbsencePeriod::getAbsencePeriods();
+    // Assert only that the number of items is the expected one,
```

Comment:
It's not that simple. I would have process the date before inserting it into the database and after, meaning I would need to add logic to the fabricator, which I would prefer to avoid as much as possible (it's used only to create data).

Besides, as I mentioned on the PR description, this is for hrabsence extension which will soon be replaced by L&A. I don't think it's worth spend time making it perfect. When we migrated from 4.7 this extensions tests were not even updated for the same reason. So I think that the test is good enough just the way it is